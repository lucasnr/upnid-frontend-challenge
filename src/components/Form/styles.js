import styled from 'styled-components';

export const Container = styled.form`
	align-items: center;
	display: flex;
	margin-bottom: 1.5rem;
	width: 20rem;

	input {
		border: none;
		border-radius: 0.125rem;
		font-size: 0.75rem;
		flex: 1;
		margin-right: 0.75rem;
		width: 100%;
		padding: 0.375rem 1rem;
	}

	button {
		background-color: #49e;
		border: none;
		border-radius: 0.125rem;
		color: #fff;
		font-size: 0.75rem;
		padding: 0.375rem 1rem;
		text-transform: uppercase;

		span {
			display: none;
		}
	}

	@media (min-width: 576px) {
		width: 30rem;

		input,
		button {
			font-size: 1rem;

			span {
				display: inline;
			}
		}
	}
`;
