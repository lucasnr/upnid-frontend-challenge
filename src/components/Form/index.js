import React from 'react';

import { Container } from './styles';

export default function Form({ onSubmit }) {
	return (
		<Container onSubmit={onSubmit}>
			<input id="name" type="text" name="name" placeholder="Nome" />
			<button type="submit">
				Iniciar<span> corrida</span>
			</button>
		</Container>
	);
}
