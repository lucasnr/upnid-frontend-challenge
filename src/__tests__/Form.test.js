import React from 'react';
import { shallow } from 'enzyme';

import Form from '../components/Form';

describe('Testing form component', () => {
	it('renders without crashing and matches snapshot', () => {
		const wrapper = shallow(<Form />);

		expect(wrapper).toMatchSnapshot();
	});

	it('executes callback on submit', () => {
		const callback = jest.fn();

		const wrapper = shallow(<Form onSubmit={callback} />);
		wrapper.simulate('submit');
		expect(callback).toHaveBeenCalled();
	});

	it('renders input with prop value', () => {
		const value = 'Some value that should be in the input';
		const wrapper = shallow(<Form value={value} />);

		expect(wrapper.find('input').props().value).toBe(value);
	});

	it('executes callback on input change', () => {
		let text;
		const value = 'Some random value';
		const callback = jest.fn((event) => {
			text = event.target.value;
		});

		const wrapper = shallow(<Form onChangeText={callback} />);
		wrapper.find('input').simulate('change', { target: { value } });

		expect(callback).toHaveBeenCalled();
		expect(text).toBe(value);
	});
});
