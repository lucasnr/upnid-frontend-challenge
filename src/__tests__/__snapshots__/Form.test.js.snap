// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing form component renders without crashing and matches snapshot 1`] = `
<styled.form>
  <input
    id="name"
    name="name"
    placeholder="Nome"
    type="text"
  />
  <button
    type="submit"
  >
    Iniciar
    <span>
       corrida
    </span>
  </button>
</styled.form>
`;
